summingbird-storm/src/main/scala/com/twitter/summingbird/storm/collector/AggregatorOutputCollector.scala

```diff
@@ -51,24 +49,25 @@ class AggregatorOutputCollector[K, V: Semigroup](
     The method is invoked to handle the flushed cache caused by
     exceeding the memoryLimit, which is called within add method.
    */
-  private def emitData[K, V](cache: Future[Traversable[(Int, CMap[K, V])]], streamId: String): List[Int] = {
+  private def emitData[K, V](cache: Future[TraversableOnce[(Int, CMap[K, V])]], streamId: String): JList[Integer] = {
     val flushedTups = Await.result(cache)
     val messageIdsTracker = streamMessageIdTracker(streamId)
-    val returns = flushedTups.toList
-      .map {
-        case (k, v) =>
-          val messageIds = messageIdsTracker.remove(k)
-          val list = List(k, v).asJava.asInstanceOf[JList[AnyRef]]
-          callEmit(messageIds, list, streamId)
-      }
-    returns.flatMap { x => { if (x != null) x.asScala.toList.asInstanceOf[List[Int]] else List[Int]() } }
+    val returns = flushedTups.map {
+      case (k, v) =>
+        val messageIds = messageIdsTracker.remove(k)
+        val list = List(k, v).asJava.asInstanceOf[JList[AnyRef]]
```

Comment:
maybe it would be better to just do:
```scala
val list = new java.util.ArrayList[AnyRef](2)
list.add(k.asInstanceOf[AnyRef])
list.add(v.asInstanceOf[AnyRef])
```